### PR TITLE
Add missing examples for MaybeUninit

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
ping @Dylan-DPC

File: library/core/src/mem/maybe_uninit.rs [hunk 1]
```diff
@@ -246,6 +246,14 @@ impl<T> MaybeUninit<T> {
     /// Note that dropping a `MaybeUninit<T>` will never call `T`'s drop code.
     /// It is your responsibility to make sure `T` gets dropped if it got initialized.
     ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::mem::MaybeUninit;
+    ///
+    /// let v = MaybeUninit::new(vec![42]);
```
Comment (pickfire) [inline library/core/src/mem/maybe_uninit.rs:254]:
```suggestion
    /// let v: MaybeUninit<Vec<u8>> = MaybeUninit::new(vec![42]);
```
Add a type for clarification.

**Review by pickfire — CHANGES_REQUESTED** [review]
Other than the mentioned change looks good to me.

File: library/core/src/mem/maybe_uninit.rs [hunk 2]
```diff
@@ -246,6 +246,14 @@ impl<T> MaybeUninit<T> {
...
+    /// ```
+    ///
     /// [`assume_init`]: MaybeUninit::assume_init
```
Comment (pickfire) [inline library/core/src/mem/maybe_uninit.rs:257]: I feel like removing these links.

File: library/core/src/mem/maybe_uninit.rs [hunk 3]
```diff
@@ -261,6 +269,14 @@ impl<T> MaybeUninit<T> {
...
     ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::mem::MaybeUninit;
+    ///
+    /// let v: MaybeUninit<String> = MaybeUninit::uninit();
+    /// ```
+    ///
     /// [type]: union.MaybeUninit.html
```
Comment (pickfire) [inline library/core/src/mem/maybe_uninit.rs:280]:
```suggestion
    /// [type]: MaybeUninit
```
I think this should work?

**Review by pickfire — APPROVED** [review]
Looks good to me.

**Comment by GuillaumeGomez** [comment]
Let's merge then, thanks for the review!

@bors: r=pickfire rollup

**Pull request merged**